src/load.ts

```diff
@@ -0,0 +1,59 @@
+import Promise from './Promise';
+
+declare var define: {
+	(...args: any[]): any;
+	amd: any;
+};
+
+export interface AMDRequire {
+	(moduleIds: string[], callback: (...modules:any[]) => void): void;
+}
+export interface NodeRequire {
+	(moduleId: string): any;
+}
+export type Require = AMDRequire | NodeRequire;
+
+export interface Load {
+	(require: Require|string, ...moduleIds: string[]): Promise<any[]>;
```

Comment:
Off the top of my head, I don't remember why I did this. Perhaps I was being clever. When I review this over the weekend, I'll see if I can come up with a better defense ;).